spec/unit/provider/package_spec.rb

```diff
@@ -711,11 +711,11 @@ class MyPackageProvider < Chef::Provider::Package
       expect { provider.run_action(:install) }.not_to raise_error
     end
 
-    it "raises an exception if an explicit version is asked for, an old version is installed, but no candidate" do
+    it "does not raise an exception if an explicit version is asked for, an old version is installed, but no candidate" do
       new_resource.version ["1.0", "6.2"]
       current_resource.version(["1.0", "6.1"])
       provider.candidate_version = ["1.0", nil]
-      expect { provider.run_action(:install) }.to raise_error(Chef::Exceptions::Package)
```

Comment:
this test is no longer correct.

if we have "6.1" installed, and we want "6.2" then we need to perform an action, but the repo lacks any version at all, so we must blow up -- if we attempt to install then we can't install 6.2.   candidate_version of nil is perfectly correct in this case, and you can't overwrite it with "6.2"